### PR TITLE
Deploy Ten Gateway as an application running inside SGX container

### PR DESCRIPTION
**Pull request by zkokelj**
### Why this change is needed

Please provide a description and a link to the underlying ticket

### What changes were made as part of this PR

Please provide a high level list of the changes made

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: .github/workflows/manual-deploy-ten-gateway-frontend.yml [hunk 1]
```diff
@@ -26,46 +26,54 @@ jobs:
...
+            GATEWAY_API_URL: 'https://obscurogateway-dev-testnet.uksouth.cloudapp.azure.com'
           - testnet_type: 'uat-testnet'
-            GATEWAY_API_URL: 'https://uat-testnet.ten.xyz'
+            GATEWAY_API_URL: 'https://obscurogateway-uat-testnet.uksouth.cloudapp.azure.com'
           - testnet_type: 'sepolia-testnet'
-            GATEWAY_API_URL: 'https://testnet.ten.xyz'
+            GATEWAY_API_URL: 'https://obscurogateway-sepolia-testnet.uksouth.cloudapp.azure.com'
     steps:
       - name: 'Print GitHub variables'
+        if: matrix.testnet_type == github.event.inputs.testnet_type
```
Comment (anixon604) [inline .github/workflows/manual-deploy-ten-gateway-frontend.yml:36]: You should be able to reduce the matrix condition checks by having a top level step - cause each matrix will iterate overall the steps then to the next matrix:
```      
        - name: 'Check if selected testnet matches matrix testnet'
        if: matrix.testnet_type != github.event.inputs.testnet_type
        run: exit 0
```

File: tools/walletextension/main/testnet.pem [hunk 2]
```diff
@@ -0,0 +1,39 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (anixon604) [inline tools/walletextension/main/testnet.pem:1]: Is it possible that we put this in a gh secret and then pull it out, https://tinyurl.com/yc78scsm

**Review by anixon604 — CHANGES_REQUESTED** [review]
Just a couple small things, mainly thinking of the pem, see comments.

File: .github/workflows/manual-deploy-ten-gateway-frontend.yml (hunk 1 above)
Comment (zkokelj) [reply .github/workflows/manual-deploy-ten-gateway-frontend.yml]: nice one, thanks 🙏 
fixed

File: tools/walletextension/main/testnet.pem (hunk 2 above)
Comment (zkokelj) [reply tools/walletextension/main/testnet.pem:1]: added that, but I think we will need to change it also for L2 deployment. The same testnet.pem file is also there.

**Comment by zkokelj** [comment]
split it into multiple smaller PRs

**Pull request closed without merging**